timely/src/progress/reachability.rs

```diff
@@ -145,8 +147,9 @@ pub struct Builder<T: Timestamp> {
 impl<T: Timestamp> Builder<T> {
 
     /// Create a new empty topology builder.
-    pub fn new() -> Self {
+    pub fn new(path: Vec<usize>) -> Self {
```

Comment:
I think this is a big abstraction change, in that `reachability.rs` used to be agnostic to the hierarchical nature of names in timely dataflow, and just did reachability tracking in a scope with no additional information. This seems to bake that in now, which .. will have to ponder whether that is a good call or not.

Another option might be to produce another implementation of `reachability.rs`, putting things behind a trait.

Not trying to be difficult, but attempting to minimize the complexity in an already-too-complicated bit of logic.

Reply:
That's a good point. I removed the path argument from from Tracker and Builder constructor. 
However, it is still added to the tracker when the logger is registered. The logging events such as "a capability is added/removed from a location" do not make sense if they cannot be tracked down to a particular location.

Let me know what you think please and whether you would still prefer going your way.